From bfe6b7286f089ef3290c0fe3500756fa2f9e4d20 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:20:36 -0400
Subject: [alsa] hda: Enable front audio jacks on one HP desktop model

Message-id: <140972823573-132-patchset-jkysela@redhat.com>
Patchwork-id: 90832
O-Subject: [RHEL 7.1 PATCH 131/442] ALSA: hda - Enable front audio jacks on one HP desktop model
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The front headphone and mic jackes on a HP desktop model (Vendor Id:
0x111d76c7 Subsystem Id: 0x103c2b17) can not work, the codec on this
machine has 8 physical ports, 6 of them are routed to rear jackes
and all of them work very well, while the remaining 2 ports are
routed to front headphone and mic jackes, but the corresponding
pin complex node are not defined correctly.

After apply this fix, the front audio jackes can work very well.

[trivial fix of enum definition by tiwai]

BugLink: https://bugs.launchpad.net/bugs/1282369
Cc: David Henningsson <david.henningsson@canonical.com>
Tested-by: Gerald Yang <gerald.yang@canonical.com>
Cc: stable@vger.kernel.org
Signed-off-by: Hui Wang <hui.wang@canonical.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: 1de7ca5e844866f56bebb2fc47fa18e090677e88)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_sigmatel.c b/sound/pci/hda/patch_sigmatel.c
index 7311bad..a2f11bf 100644
--- a/sound/pci/hda/patch_sigmatel.c
+++ b/sound/pci/hda/patch_sigmatel.c
@@ -83,6 +83,7 @@ enum {
  STAC_DELL_M6_BOTH,
  STAC_DELL_EQ,
  STAC_ALIENWARE_M17X,
+ STAC_92HD89XX_HP_FRONT_JACK,
  STAC_92HD73XX_MODELS
 };
 
@@ -1795,6 +1796,12 @@ static const struct hda_pintbl intel_dg45id_pin_configs[] = {
  {}
 };
 
+static const struct hda_pintbl stac92hd89xx_hp_front_jack_pin_configs[] = {
+ { 0x0a, 0x02214030 },
+ { 0x0b, 0x02A19010 },
+ {}
+};
+
 static void stac92hd73xx_fixup_ref(struct hda_codec *codec,
        const struct hda_fixup *fix, int action)
 {
@@ -1913,6 +1920,10 @@ static const struct hda_fixup stac92hd73xx_fixups[] = {
  [STAC_92HD73XX_NO_JD] = {
   .type = HDA_FIXUP_FUNC,
   .v.func = stac92hd73xx_fixup_no_jd,
+ },
+ [STAC_92HD89XX_HP_FRONT_JACK] = {
+  .type = HDA_FIXUP_PINS,
+  .v.pins = stac92hd89xx_hp_front_jack_pin_configs,
  }
 };
 
@@ -1973,6 +1984,8 @@ static const struct snd_pci_quirk stac92hd73xx_fixup_tbl[] = {
         "Alienware M17x", STAC_ALIENWARE_M17X),
  SND_PCI_QUIRK(PCI_VENDOR_ID_DELL, 0x0490,
         "Alienware M17x R3", STAC_DELL_EQ),
+ SND_PCI_QUIRK(PCI_VENDOR_ID_HP, 0x2b17,
+    "unknown HP", STAC_92HD89XX_HP_FRONT_JACK),
  {} /* terminator */
 };
 
-- 
1.7.1